### ISSUE TITLE
Rewrite seems to be broken

### ISSUE THREAD

**Issue by jwj61** [LMFDB/lmfdb#3571]
Trying to rewrite the number field database, it seemed to dump it, and then failed.  I tried again using the identify function for the modification:

from lmfdb import db

def nullmod(ent):
    return ent

db.nf_fields.rewrite(nullmod,searchfile='/scratch/home/jj/nf-lmfdb/DELETE.ME',ex
trafile='/scratch/home/jj/nf-lmfdb/EXTRA-DELETE.ME')

produces

All records dumped in 42454.004 secs
Reloading nf_fields...
Traceback (most recent call last):
  File "./modify.py", line 152, in <module>
    db.nf_fields.rewrite(nullmod,searchfile='/scratch/home/jj/nf-lmfdb/DELETE.ME',extrafile='/scratch/home/jj/nf-lmfdb/EXTRA-DELETE.ME')
  File "/scratch/home/jj/lmfdb/lmfdb/backend/database.py", line 3134, in rewrite
    self.reload(searchfile.name, extrafile.name, resort=resort, reindex=reindex, restat=restat, commit=commit, log_change=False, **kwds)
  File "/scratch/home/jj/lmfdb/lmfdb/backend/database.py", line 3716, in reload
    addid, counts[table] = self._copy_from(filename, tmp_table, cols, header, kwds)
  File "/scratch/home/jj/lmfdb/lmfdb/backend/database.py", line 905, in _copy_from  
    cur.copy_from(F, table, columns=columns, **kwds)
psycopg2.errors.InvalidTextRepresentation: invalid input syntax for integer: "0|1.1.1.1|1|0|False|1|1|1|1|1|1|[]|False|1.0|1|{}|{0,1}|0|1|{}|{}|2|N"
CONTEXT:  COPY nf_fields_tmp, line 1, column id: "0|1.1.1.1|1|0|False|1|1|1|1|1|1|[]|False|1.0|1|{}|{0,1}|0|1|{}|{}|2|N"

**Comment by roed314** [comment]
Yikes!  That's a long time to wait for a rewrite.  I've created a new table where we can replicate the problem: `test_rewrite`.  
```
db.test_rewrite.rewrite(nullmod, searchfile='/home/roe/DELETE_ME.txt')
All records dumped in 0.006 secs
Reloading test_rewrite...
...
InvalidTextRepresentation: invalid input syntax for integer: "0|1.2.3|N"
CONTEXT:  COPY test_rewrite_tmp, line 1, column id: "0|1.2.3|N"
```

The issue is that there's some problem with backslash not being inserted appropriately, so we get `N` instead of `\N`.  I can't actually look at the file, since it's deleted in the `finally` block of rewrite.

I'll try to track down where we need to add a backslash soon....

**Comment by roed314** [comment]
I was wrong; the issue was actually a missing `sep` keyword from the call to the cursor's `copy_from` method.  It's a result of a recent change from tabs to `|` as the default separator.

**Comment by jwj61** [comment]
Fixed by #3572
